File: C_Programming/Assignment/POC/perfectfun.c
#include<stdio.h>
void perfect();
void main()
{
   perfect();
}
void perfect()
{
    int n,sum=0;
    printf("\n Enter the value:");
    scanf("%d",&n);
    int i=1;
    while(i<=n/2)
    {
        if(n%i==0)
        {
            sum=sum+i;
            
        }
      i++;
    }
    if(sum==n)
    {
      printf("The no is perfect");
    }
    
  
      else
      {
        printf("The no is not perfect");
      }

}

//output:Enter the value:24
// The no is not perfect
// Enter the value:28
// The no is perfect

File: Java_Programs/Assignment5Type(3)/perfectfun.java
package Assignment5Type3;

public class perfectfun {

	public static void main(String[] args) {
		int n=28;
//	    printf("\n Enter the value:");
//	    scanf("%d",&n);
	    isperfect(n);

	}

	public static void isperfect(int n)
	{
	  int sum=0;
	    int i=1;
	    while(i<=n/2)
	    {
	        if(n%i==0)
	        {
	            sum=sum+i;
	            
	        }
	      i++;
	    }
	    if(sum==n)
	    {
	     System.out.printf("The number is perfect");
	    }
	    
	  
	      else
	      {
	       System.out.printf("The number is not perfect");
	      }

	    }

}
